processing/src/main/java/org/apache/carbondata/processing/loading/parser/impl/MapParserImpl.java

```diff
@@ -44,12 +44,13 @@ public MapParserImpl(String delimiter, String nullFormat, String keyValueDelimit
         String[] split = pattern.split(value, -1);
         if (ArrayUtils.isNotEmpty(split)) {
           ArrayList<Object> array = new ArrayList<>();
-          Set<Object> set = new HashSet<>();
+          Map<Object, String> map = new HashMap<>();
           for (int i = 0; i < split.length; i++) {
             Object currKey = split[i].split(keyValueDelimiter)[0];
-            if (set.add(currKey)) {
-              array.add(child.parse(split[i]));
-            }
+            map.put(currKey, split[i]);
+          }
+          for (Map.Entry<Object, String> entry : map.entrySet()) {
```

Comment:
This may change the order of user input. You can use LinkedHashMap to maintain the insertion order.

Reply:
I suggest it would be better if we maintain a sorted order inside the map.

Reply:
@manishnalla1994 what sorted order are you referring to? Please check the behaviour of hive whether they maintain the insertion order or not.

Reply:
Here sorted order means the keys will be sorted.

Reply:
@ravipesala hive also maintains the sorted order of keys, as is the current implementation here.